denops/ddc/sources/around.ts

```diff
@@ -39,22 +39,21 @@ export class Source extends BaseSource {
     const pageSize = 500;
     const p = sourceParams as unknown as Params;
     const maxSize = p.maxSize;
-    const currentLine = (await denops.call("line", ".")) as number;
+    const currentLine = await fn.line(denops, ".");
     const minLines = Math.max(1, currentLine - maxSize);
     const maxLines = Math.min(
-      (await denops.call("line", "$")) as number,
+      await fn.line(denops, "$"),
       currentLine + maxSize,
     );
-    const pages = (await Promise.all(
-      imap(
-        splitPages(minLines, maxLines, pageSize),
-        ([start, end]: [number, number]) => denops.call("getline", start, end),
-      ),
-    )) as string[][];
-    const lines = pages.flatMap((page) => page);
-
-    const candidates: Candidate[] = allWords(lines).map((word) => ({ word }));
-    return candidates;
+    const ps = await batch(denops, (helper) => {
+      for (const [s, e] of splitPages(minLines, maxLines, pageSize)) {
+        fn.getline(helper, s, e);
+      }
+    }) as string[][];
```

Comment:
As @octaltree and I said in vim-jp Slack, probably the following code is more efficient while RPC cost a bit (we have not measured though)

I keep previous implementation while I thought this `around` source is a kind of sample code.

```typescript
const cs: Candidate[] = [...new Set(
  allWords(await fn.getline(denops, 1, '$')).map((word) => ({ word }))
)];
```

Reply:
If the lines are huge, it may block your Vim.
I think to avoid block is important the performance.